src/components/Checkbox/__tests__/Checkbox.test.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import {render, screen} from '@testing-library/react';
+import {Checkbox, CheckboxSize} from '../Checkbox';
+
+const qaId = 'checkbox-component';
+
+describe('Checkbox', () => {
+    test('render checkbox by default', () => {
+        render(<Checkbox />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).toBeVisible();
+        expect(checkbox).not.toBeDisabled();
+        expect(checkbox).not.toBeChecked();
+    });
+
+    test.each(new Array<CheckboxSize>('m', 'l'))('render with given "%s" size', (size) => {
+        render(<Checkbox size={size} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveClass(`yc-checkbox_size_${size}`);
+    });
+
+    test('disabled when disabled=true prop is given', () => {
+        render(<Checkbox disabled={true} />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).toBeDisabled();
+    });
+
+    test('not disabled when disabled=false prop is given', () => {
+        render(<Checkbox disabled={false} />);
+        const checkbox = screen.getByRole('checkbox');
+
+        expect(checkbox).not.toBeDisabled();
+    });
+
+    test('render with indeterminate icon', () => {
+        render(<Checkbox indeterminate={true} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        // eslint-disable-next-line testing-library/no-node-access
+        expect(component.querySelector('svg path')?.getAttribute('d')).toEqual('M4 7h9v3H4z');
+    });
+
+    test('set given title to label', () => {
+        const title = 'Some title';
+
+        render(<Checkbox title={title} />);
+        const label = screen.getByTitle(title);
+
+        expect(label).toBeVisible();
+    });
+
+    test('show given content', () => {
+        const content = 'Some content';
+
+        render(<Checkbox content={content} />);
+        const text = screen.getByText(content);
```

Comment:
I think that `getByLabel` is more robust

Reply:
Replaced by `getByLabelText`

Reply:
Restored to `getByText` - we need to check the node associated with text, not the checkbox itself